sources/settings-extensions/oci-hooks/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+/// Settings related to host-provided OCI Hooks
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use model_derive::model;
+use std::convert::Infallible;
+
+#[model(impl_default = true)]
+pub struct OciHooksSettingsV1 {
+    log4j_hotpatch_enabled: bool,
+}
+
+type Result<T> = std::result::Result<T, Infallible>;
+
+impl SettingsModel for OciHooksSettingsV1 {
+    type PartialKind = Self;
+    type ErrorKind = Infallible;
+
+    fn get_version() -> &'static str {
+        "v1"
+    }
+
+    fn set(_current_value: Option<Self>, _target: Self) -> Result<()> {
+        // Set anything that parses as OciHooksSettingsV1.
+        Ok(())
+    }
+
+    fn generate(
+        existing_partial: Option<Self::PartialKind>,
+        _dependent_settings: Option<serde_json::Value>,
+    ) -> Result<GenerateResult<Self::PartialKind, Self>> {
+        Ok(GenerateResult::Complete(
+            existing_partial.unwrap_or_default(),
+        ))
+    }
+
+    fn validate(_value: Self, _validated_settings: Option<serde_json::Value>) -> Result<()> {
+        // Validate anything that parses as OciHooksSettingsV1.
```

Comment:
I know this has been copied from another settings extension, but I think it could be more clear.

```suggestion
        // OciHooksSettingsV1 is validated during deserialization.
```

Reply:
Makes sense, will update.